floodviz/hydrograph_utils.py

```diff
@@ -57,12 +61,28 @@ def req_hydrodata(sites, start_date, end_date, url_top):
         url_top - URL endpoint for the nwis web service
     
     RETURNS:
-        returns a dictonary with the requested data from the nwis service 
+        returns a list of one dictonary with the requested data for
+        all series from the nwis service 
     
     """
-    sites_string = ','.join(sites)
-    url =  url_top +'iv/?site=' + sites_string + '&startDT=' + \
+    if len(sites) is not 0 and start_date and end_date and url_top:
+        sites = [str(site) for site in sites]
+        sites_string = ','.join(sites)
+        url =  url_top +'iv/?site=' + sites_string + '&startDT=' + \
               start_date + '&endDT=' + end_date + '&parameterCD=00060&format=json'
-    r = requests.get(url)
-    if r.status_code is 200:
-        return r.json()['value']['timeSeries']
\ No newline at end of file
+
+        try:
+            r = requests.get(url)
+            if r.status_code is 200:
+                return r.json()['value']['timeSeries']
+            else:
+                print('\n - Bad Request -\n')
+
+        except (requests.exceptions.MissingSchema) as e:
+            print('\n - Malformed URL - \n')
+
+    else:
+        print('\nConfig Varibles Empty\n')
+    
+    return None
```

Comment:
You don't need to change this now, but try to have one return statement for a function.